corehq/util/tests/test_doc_processor.py

```diff
@@ -471,12 +454,6 @@ def test_multiple_runs_no_skip(self):
         self._test_processor(4, list(range(4)))
         self._test_processor(0, [])
 
-    def test_multiple_runs_with_skip(self):
-        with self.assertRaises(TooManyRetries):
-            self._test_processor(3, list(range(3)), skip_docs=['bar-3'])
-
-        self._test_processor(1, [3])
```

Comment:
Is this test not still useful? It fails now because there's no exception being raised. Is the skip logic tested elsewhere?

Reply:
I don't think so because it relies on the retry mechanism. The first `_test_processor` call always skips/fails on `bar-3`, hence `TooManyRetries`. The second `_test_processor` call does not skip that doc on retry, so is able to process it successfully.

Reply:
I think the skip logic is tested here

https://github.com/dimagi/commcare-hq/blob/7db51c5ca70978abfc2ab0fc2c4f32970e437c48/corehq/util/tests/test_doc_processor.py#L478-L479

Although the name of that test makes me thing it somehow depends on the retry mechanism too. Wonder how that's passing. I'll revisit that.

Reply:
https://github.com/dimagi/commcare-hq/pull/29743/commits/ffc5ebc65f49751e1b295205088dc02b6c558613

Answering the question from my previous comment, that is testing `BulkDocProcessor`, which overrides `_process_doc` and handles skips in a different way (not using the retry mechanism that was removed here).

Reply:
Thanks for the follow up